### PR TITLE
Add/28568 dashboard widget - setup

### LINKED ISSUE #28568: WooCommerce Status Widget: Add Store Setup State
The Start/Manage team is going to be looking at updating some of the WooCommerce wp-admin dashboard widgets, and this is the first of a few issues that Isotope will be looking into in the new year ( or possibly sooner! ).

Dashboard ‹ bend outdoors — WordPress 2020-12-11 17-03-58

One of the changes we would like to make to the Status Widget on the wp-admin dashboard, is to make it more useful for brand-new stores that are still in the process of getting things setup and configured.

On Woo pages, we display the "Store Setup" link in the Activity Panel area if a store has remaining setup items to perform, but this visual nudge is not shown on the wp-admin dashboard page. As such, the idea is to utilize the existing Status Widget to remind managers of the site that they have remaining setup tasks to do.

We implemented a similar solution in Calypso on wordpress.com in https://github.com/Automattic/wp-calypso/pull/44547

Acceptance Criteria:

- [ ] When a store is still in setup mode, as determined by the [logic that shows the Store Setup activity panel](https://github.com/woocommerce/woocommerce-admin/blob/main/client/header/activity-panel/index.js#L147-L151) the Status Widget should not show the standard stats/links, rather it should show the new design ( TBD ) informing users that their store has remaining setup tasks.
- [ ] Once the setup task list has been completed, or hidden, the Status Widget should render the standard state/content seen in the screenshot above.

cc @elizaan36 as this is an issue we will need some design assistance with.

### PR DESCRIPTION
**Pull request by moon0326**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #28568 

This PR adds "WooCommerce Setup" widget to promote users to finish the onboarding tasks. The widget shows only when there are unfinished tasks.

This PR also hides the existing "WooCommerce Status" widget until the store owner completes the onboarding tasks.

### How to test the changes in this Pull Request:

1. Start with a fresh installation of WooCoommerce and don't start onboarding wizard yet.
2. Navigate to WordPress Dashboard and confirm that you see a new widget with the title "WooCommerce Setup", but "WooCommece Status" widget is hidden. 
3. Confirm that the widget shows "Step 0 of 5" (the total # of steps can be 6 if you have a payment plugin available)
4. Start working on onboarding tasks one by one.
5. As you finish onboarding tasks, check back and confirm that step # increases. 
6. The "WooCommerce Setup" widget should not be displayed once you complete the onboarding tasks, but "WooCommerce Status" widget should be displayed again.

### Screenshots
Screen Shot 2021-02-18 at 2 50 49 PM

## Note

I've noticed that we have the onboarding tasks defined in multiple places now. It might not be a big deal, but we have to update two places when we add a new onboarding task. It would be nice if we have an API where we can get a list of onboarding tasks. I will try to see if it's worth working on that during the upcoming cooldown period. 

[tasks defined in js
](https://github.com/woocommerce/woocommerce-admin/blob/main/client/task-list/tasks.js)

[tasks defined in this PR](https://github.com/woocommerce/woocommerce/pull/29174/files#diff-857cada53ba223723d6caaabeaddb6909539ac1a748781e4718d6f36b9c03172R27)

### Changelog entry

> Enhancement - Add a new dashboard widget to promote users to finish onboarding tasks.

### CODE REVIEW

**Comment by elizaan36** [comment]
hey @moon0326 wow, this looks so great 👏

Wanted to mention one small thing - there's a new `badge` component that you could use here for the step count (step 1 of 5, etc). It's a more rounded pill shape, less padding, and lighter grey.

image

**Comment by moon0326** [comment]
> hey @moon0326 wow, this looks so great 👏
> 
> Wanted to mention one small thing - there's a new `badge` component that you could use here for the step count (step 1 of 5, etc). It's a more rounded pill shape, less padding, and lighter grey.
> 
> image

Hey @elizaan36,

Oh well..I wish I could use a component. React isn't available on the dashboard...so I had to work on it myself. I'm going to update the shape and color according to the component. Thank you!

**Comment by moon0326** [comment]
> hey @moon0326 wow, this looks so great 👏
> 
> Wanted to mention one small thing - there's a new `badge` component that you could use here for the step count (step 1 of 5, etc). It's a more rounded pill shape, less padding, and lighter grey.
> 
> image

@elizaan36 Could you link me to the component page? I found `badge` component in WooCommerce Admin storybook, but it does not look the same. I'm trying to copy the exact roundness and padding values.

File: includes/admin/views/html-admin-dashboard-setup.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		</svg>
+		<span><?php echo esc_html_e( 'Step', 'woocommerce' ); ?> <?php echo esc_html( $completed_tasks_count ); ?> <?php echo esc_html_e( 'of', 'woocommerce' ); ?> <?php echo esc_html( $tasks_count ); ?></span>
+	</span>
+
+	<div class="description">
+		<div>
+			<?php echo esc_html_e( 'You\'re almost there! Once you complete store setup you can start receiving orders.', 'woocommerce' ); ?>
+			<div><a href='<?php echo esc_attr( $button_link ); ?>' class='button button-primary'><?php echo esc_html_e( 'Start selling', 'woocommerce' ); ?></a></div>
+		</div>
+		<img src="<?php echo esc_url( WC()->plugin_url() ); ?>/assets/images/dashboard-widget-setup.png"" />
```
Comment (becdetat) [inline includes/admin/views/html-admin-dashboard-setup.php:26]: What is this image used for? Also it looks like there is a doubled quotation mark at the end of the `src` attribute.

Comment (moon0326) [reply includes/admin/views/html-admin-dashboard-setup.php:26]: Thank you for catching it. I will remove the extra double quote. Interesting my IDE did not catch that..

The image is for the person on the right side.  

This guy :) 

Screen Shot 2021-02-22 at 6 15 18 PM

File: includes/admin/class-wc-admin-dashboard-setup.php [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+				$this->init();
+			}
+		}
+
+		/**
+		 * Hook meta_box
+		 */
+		public function init() {
+			add_meta_box(
+				'wc_admin_dasbharod_setup',
```
Comment (becdetat) [inline includes/admin/class-wc-admin-dashboard-setup.php:82]: typo here

File: includes/admin/class-wc-admin-dashboard-setup.php [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+		/**
+		 * WC_Admin_Dashboard_Setup constructor.
+		 */
+		public function __construct() {
+			if ( $this->should_display_widget() ) {
+				$this->populate_general_tasks();
+				$this->populate_payment_tasks();
+				$this->completed_tasks_count = $this->get_completed_tasks_count();
+				$this->init();
```
Comment (becdetat) [inline includes/admin/class-wc-admin-dashboard-setup.php:73]: `init()` looks like it could just be inlined here, rather than have a seperate function

**Comment by becdetat** [comment]
Everything is looking good, it works really well, I just had a couple of comments

File: includes/admin/class-wc-admin-dashboard-setup.php (hunk 3 above)
Comment (moon0326) [reply includes/admin/class-wc-admin-dashboard-setup.php:73]: Makes sense. I will make an update 👍

File: includes/admin/class-wc-admin-dashboard-setup.php (hunk 2 above)
Comment (moon0326) [reply includes/admin/class-wc-admin-dashboard-setup.php:82]: Thank you :)

**Comment by moon0326** [comment]
> Everything is looking good, it works really well, I just had a couple of comments

Thank you! Made the requested changes in https://github.com/woocommerce/woocommerce/pull/29174/commits/42606169854e244effe511ae4369fe4e89049b78

**Comment by elizaan36** [comment]
> @elizaan36 Could you please link me to the component page? I found badge component in WooCommerce Admin storybook, but it does not look the same. I'm trying to copy the exact roundness and padding values.

No problem, I shared the Figma link to the most recent component design, though _I think_ the StoryBook component should have the same padding.

**Comment by moon0326** [comment]
> > @elizaan36 Could you please link me to the component page? I found badge component in WooCommerce Admin storybook, but it does not look the same. I'm trying to copy the exact roundness and padding values.
> 
> No problem, I shared the Figma link to the most recent component design, though _I think_ the StoryBook component should have the same padding.

Thank you! I've tried my best to match the design in Figma.

This is the result.

change copy

File: includes/admin/class-wc-admin-dashboard.php [hunk 4]
```diff
@@ -24,7 +24,7 @@ class WC_Admin_Dashboard {
 		 */
 		public function __construct() {
 			// Only hook in admin parts if the user has admin access.
-			if ( current_user_can( 'view_woocommerce_reports' ) || current_user_can( 'manage_woocommerce' ) || current_user_can( 'publish_shop_orders' ) ) {
+			if ( $this->should_display_widget() && ( current_user_can( 'view_woocommerce_reports' ) || current_user_can( 'manage_woocommerce' ) || current_user_can( 'publish_shop_orders' ) ) ) {
```
Comment (becdetat) [inline includes/admin/class-wc-admin-dashboard.php:27]: all of this logic could be moved into `$this->should_display_widget()` and simplified. Not required though.

**Review by becdetat — DISMISSED** [review]
Everything looks good, I added one minor optional comment but I'll approve this. Nice work!

File: includes/admin/class-wc-admin-dashboard.php (hunk 4 above)
Comment (moon0326) [reply includes/admin/class-wc-admin-dashboard.php]: Updated in https://github.com/woocommerce/woocommerce/pull/29174/commits/d013d7f7659245960e051bcb4138bf36b210e2ec

**Comment by moon0326** [comment]
> Everything looks good, I added one minor optional comment but I'll approve this. Nice work!

Thank you! I'm gonna run a few more tests and merge soon 👍

**Review by becdetat — APPROVED** [review]

**Pull request merged**